app/src/main/java/org/schabi/newpipe/player/ui/PlayerUiList.kt

```diff
@@ -0,0 +1,124 @@
+package org.schabi.newpipe.player.ui
+
+import org.schabi.newpipe.util.GuardedByMutex
+import java.util.Optional
+
+class PlayerUiList(vararg initialPlayerUis: PlayerUi) {
+    var playerUis = GuardedByMutex(mutableListOf<PlayerUi>())
```

Comment:
[`Collections.synchronizedList()`](https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#synchronizedList-java.util.List-) could be used instead, which would remove the need for a separate mutex class.

```suggestion
    var playerUis = Collections.synchronizedList(mutableListOf<PlayerUi>())
```